pkg/dynamicinstrumentation/ebpf/ebpf.go

```diff
@@ -0,0 +1,207 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package ebpf provides utility for setting up and instrumenting the bpf code
+// used by dynamic instrumentation
+package ebpf
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"text/template"
+	"time"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/link"
+
+	"github.com/DataDog/datadog-agent/pkg/dynamicinstrumentation/diagnostics"
+	"github.com/DataDog/datadog-agent/pkg/dynamicinstrumentation/ditypes"
+	ddebpf "github.com/DataDog/datadog-agent/pkg/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/ebpf/bytecode/runtime"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var (
+	bpffs = "/sys/fs/bpf" //TODO: Check via `mount(2)`
+)
+
+var (
+	globalTempDirPath string
+	globalHeadersPath string
+)
+
+// SetupEventsMap creates the ringbuffer which all programs will use for sending output
+func SetupEventsMap() error {
+	var err error
+	events, err := ebpf.NewMap(&ebpf.MapSpec{
+		Name:       "events",
+		Type:       ebpf.RingBuf,
+		MaxEntries: 1 << 24,
```

Comment:
You probably want to choose a better default size since that memory is pre-allocated. It is also best to make this configurable.

Reply:
I was thinking that this was going to be one of the knobs we change once we have actual benchmarks (Alex is working on this) since we don't want to optimize too early. But still, how would you suggest we find a better default size?

Reply:
We provide ring buffer utilization metrics via `datadog.system_probe.ebpf.perf.usage_pct`. This can be used to estimate an appropriate size given some workload.